sbt-ash/src/main/scala/org/virtuslab/ash/AkkaSerializationHelperPlugin.scala

```diff
@@ -0,0 +1,91 @@
+package org.virtuslab.ash
+
+import sbt.Def
+import sbt.Keys._
+import sbt._
+
+object AkkaSerializationHelperPlugin extends AutoPlugin {
+
+  object autoImport extends AkkaSerializationHelperKeys {
+    val baseAshSettings: Seq[Def.Setting[_]] = AkkaSerializationHelperPlugin.baseAshSettings
+  }
+  import autoImport.{baseAshSettings => _, _}
+
+  val circeAkkaSerializer = component("circe-akka-serializer")
+
+  val ashAnnotation = component("ash-annotation")
+
+  override def projectSettings: Seq[Def.Setting[_]] = baseAshSettings ++ additionalSettings
+
+  override def globalSettings: Seq[Def.Setting[_]] = Nil
+
+  lazy val additionalSettings: Seq[Def.Setting[_]] = Seq(
+    libraryDependencies ++= Seq(
+        ashAnnotationLibrary.value,
+        compilerPlugin(ashDumpPersistenceSchemaCompilerPlugin.value),
+        compilerPlugin(ashRegistrationCheckerCompilerPlugin.value),
+        compilerPlugin(ashSerializabilityCheckerCompilerPlugin.value)),
+    Compile / scalacOptions ++= Seq(
+        s"-P:dump-persistence-schema-plugin:${(ashDumpPersistenceSchemaCompilerPlugin / ashCompilerPluginCacheDirectory).value}",
+        s"-P:registration-checker-plugin:${(ashRegistrationCheckerCompilerPlugin / ashCompilerPluginCacheDirectory).value}"),
+    cleanFiles ++= Seq(
+        (ashDumpPersistenceSchemaCompilerPlugin / ashCompilerPluginCacheDirectory).value / "dump-persistence-schema-cache",
+        (ashRegistrationCheckerCompilerPlugin / ashCompilerPluginCacheDirectory).value / "registration-checker-plugin-cache.csv"),
+    Compile / scalacOptions ++= (Compile / ashScalacOptions).value,
+    Test / scalacOptions --= (Compile / ashScalacOptions).value,
+    Test / scalacOptions ++= (Test / ashScalacOptions).value)
+
+  lazy val baseAshSettings: Seq[Def.Setting[_]] = Seq(
+      ashDumpPersistenceSchemaCompilerPlugin := component("dump-persistence-schema-compiler-plugin"),
+      ashRegistrationCheckerCompilerPlugin := component("registration-checker-compiler-plugin"),
+      ashSerializabilityCheckerCompilerPlugin := component("serializability-checker-compiler-plugin"),
+      ashAnnotationLibrary := ashAnnotation,
+      ashCompilerPluginCacheDirectory := target.value,
+      ashCompilerPluginEnable := true,
+      ashCompilerPluginVerbose := false,
+      DumpPersistenceSchema.dumpPersistenceSchemaTask(ashDumpPersistenceSchema),
+      ashDumpPersistenceSchema / ashDumpPersistenceSchemaOutputFile :=
+        new File(
+          (ashDumpPersistenceSchema / ashDumpPersistenceSchemaOutputDirectoryPath).value) / (ashDumpPersistenceSchema / ashDumpPersistenceSchemaOutputFilename).value,
+      ashDumpPersistenceSchema / ashDumpPersistenceSchemaOutputFilename := s"${name.value}-dump-persistence-schema-${version.value}.yaml",
+      ashDumpPersistenceSchema / ashDumpPersistenceSchemaOutputDirectoryPath := target.value.getPath) ++
+    Seq(Compile, Test).flatMap(ashScalacOptionsInConfig)
```

Comment:
As discussed, `ashDumpPersistenceSchemaCompilerPlugin` should probably be disabled in `Test` by default (unlike the other two compiler plugins)